woocommerce/woocommerce
# Add multiselection to the Variations table under Variations tab

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/39791



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and click `Add variation options` button from the `Variations` tab
4. Add at least one Variation attribute and at least one value. Then click `Add` button from the `Add variation options` modal.
5. Wait until the variations are generated
6. A list a variation should be shown depending on the options selected
7. In the Variations table each variation row should has a checkbox to select it
8. In the header of the table a another checkbox should be shown
9. When not all variations are selected, the header checkbox should be indeterminate
10. If all variations are selected then the header checkbox should be checked
11. At the header `Select all` and `Clear selection` buttons should be shown
12. If all variations are selected the `Select all` button should be disabled
13. If non variations are selected the `Clear selection` button should be disabled

<img width="714" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/c8fe605c-ddb0-492d-8de5-f3ca50df9f8b">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
